jetty-http2/http2-client/src/main/java/org/eclipse/jetty/http2/client/HTTP2ClientConnectionFactory.java

```diff
@@ -98,12 +101,42 @@ public void onOpen()
             Map<Integer, Integer> settings = listener.onPreface(getSession());
             if (settings == null)
                 settings = new HashMap<>();
-            settings.computeIfAbsent(SettingsFrame.INITIAL_WINDOW_SIZE, k -> client.getInitialStreamRecvWindow());
-            settings.computeIfAbsent(SettingsFrame.MAX_CONCURRENT_STREAMS, k -> client.getMaxConcurrentPushedStreams());
 
-            Integer maxFrameLength = settings.get(SettingsFrame.MAX_FRAME_SIZE);
-            if (maxFrameLength != null)
-                getParser().setMaxFrameLength(maxFrameLength);
+            // Here we want to populate any settings to send to the server
+            // that have a different default than what prescribed by the RFC.
+            // Changing the configuration is done when the SETTINGS is sent.
+
+            settings.compute(SettingsFrame.HEADER_TABLE_SIZE, (k, v) ->
+            {
+                if (v == null)
+                {
+                    v = client.getMaxDynamicTableSize();
```

Comment:
The value in the client is the per-connection _initial_ value.
After the connection is established, the connection specific value may be changed by SETTINGS frames, that apply to that specific connection.
A new connection gets again the initial value from the client.
While rare, there may be cases where different connections may need different configurations.

I don't see much value in `Http2Configuration` interface, as we never need to use it polymorphically.

If it is a class/record, it will be partially shared between client and server, because both have specific configurations that don't apply to the other (e.g. max concurrent streams vs max pushed streams, enable connect only valid on server, etc.).

Also, the other components (such as parser and generator) will only use some of the values, and we would need to clone the configuration to keep track of connection-specific values vs connection initial values (or duplicate the relevant component configuration in the component itself).
I'm not opposed to it, and we have actually taken that path in http3 with `HTTP3Configuration` and `QuicConfiguration`, but for http2 feels like a lot of work for little gain and still the need to maintain all the setters for backward compatibility.

Perhaps for 12, but again breaking backwards compatibility (by removing all the setters)?

Reply:
OK in regards to no Http2Configuration class.

But I still find it confusing to see `client.getMaxDynamicTableSize()`, here, but also in the constructor for the generator above.

If this is the initial value, then where does the canonical current value live?  Surely that should only be initialised once from the `client` and then dynamically managed after that?